tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraEncryptedEmailDetailedViewGetMethodContract.scala

```diff
@@ -0,0 +1,1003 @@
+import com.linagora.tmail.james.jmap.model.EncryptedEmailGetRequest
+import io.netty.handler.codec.http.HttpHeaderNames.ACCEPT
+import io.restassured.RestAssured.{`given`, requestSpecification}
+import io.restassured.http.ContentType.JSON
+import net.javacrumbs.jsonunit.assertj.JsonAssertions.assertThatJson
+import net.javacrumbs.jsonunit.core.Option
+import org.apache.http.HttpStatus
+import org.apache.james.GuiceJamesServer
+import org.apache.james.jmap.core.ResponseObject.SESSION_STATE
+import org.apache.james.jmap.core.UuidState.INSTANCE
+import org.apache.james.jmap.http.UserCredential
+import org.apache.james.jmap.rfc8621.contract.Fixture.{ACCEPT_RFC8621_VERSION_HEADER, ACCOUNT_ID, ANDRE, ANDRE_ACCOUNT_ID, ANDRE_PASSWORD, BOB, BOB_PASSWORD, DOMAIN, authScheme, baseRequestSpecBuilder}
+import org.apache.james.mailbox.MessageManager.AppendCommand
+import org.apache.james.mailbox.model.{MailboxConstants, MailboxPath, MessageId}
+import org.apache.james.mime4j.dom.Message
+import org.apache.james.mime4j.message.{BodyPartBuilder, MultipartBuilder}
+import org.apache.james.modules.MailboxProbeImpl
+import org.apache.james.utils.DataProbeImpl
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.{BeforeEach, Test}
+import play.api.libs.json.{JsString, Json}
+
+import java.nio.charset.StandardCharsets
+
+object LinagoraEncryptedEmailDetailedViewGetMethodContract {
+  val MESSAGE: Message = Message.Builder.of
+    .setSubject("test")
+    .setSender(BOB.asString)
+    .setFrom(BOB.asString)
+    .setTo(BOB.asString)
+    .setBody("test mail", StandardCharsets.UTF_8)
+    .build
+
+  val MESSAGE_HAS_ATTACHMENTS: Message = Message.Builder.of()
+    .setBody(MultipartBuilder.create("mixed")
+      .addBodyPart(BodyPartBuilder.create
+        .setBody("attachment content", "plain", StandardCharsets.UTF_8)
+        .setContentDisposition("attachment"))
+      .addBodyPart(BodyPartBuilder.create()
+        .setBody("text content", "plain", StandardCharsets.UTF_8))
+      .addBodyPart(BodyPartBuilder.create
+        .setBody("<b>html</b> content", "html", StandardCharsets.UTF_8))
+      .build)
+    .build
+
+  val MESSAGE_PREVIEW: String = "test mail"
+  val MESSAGE_HTML: String = "test mail"
+  val BOB_INBOX_PATH: MailboxPath = MailboxPath.inbox(BOB)
+}
+
+trait LinagoraEncryptedEmailDetailedViewGetMethodContract {
+
+  @BeforeEach
+  def setUp(server: GuiceJamesServer): Unit = {
+    server.getProbe(classOf[DataProbeImpl])
+      .fluent()
+      .addDomain(DOMAIN.asString)
+      .addUser(BOB.asString(), BOB_PASSWORD)
+
+    requestSpecification = baseRequestSpecBuilder(server)
+      .setAuth(authScheme(UserCredential(BOB, BOB_PASSWORD)))
+      .addHeader(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .build()
+
+    server.getProbe(classOf[MailboxProbeImpl])
+      .createMailbox(BOB_INBOX_PATH)
+
+    uploadPublicKey()
+  }
+
+  def randomMessageId: MessageId
+
+  private def uploadPublicKey(): Unit = {
```

Comment:
We could share that accross tests by moving it to a Fixture (Don't repeat yourself)